src/app/workspace/organization/detail/payments-panel/payments-list/payments-list.component.ts

```diff
@@ -17,18 +17,68 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Component, OnInit } from '@angular/core';
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {OrganizationModel} from "../../../../../core/store/organizations/organization.model";
+import {Subscription} from "rxjs/Subscription";
+import {Store} from "@ngrx/store";
+import {Router} from "@angular/router";
+import {I18n} from "@ngx-translate/i18n-polyfill";
+import {AppState} from "../../../../../core/store/app.state";
+import {selectOrganizationByWorkspace} from "../../../../../core/store/organizations/organizations.state";
+import {isNullOrUndefined} from "util";
+import {filter} from "rxjs/operators";
+import {PaymentModel} from "../../../../../core/store/organizations/payment/payment.model";
+import {selectPaymentsByWorkspaceSorted} from "../../../../../core/store/organizations/payment/payments.state";
+import {PaymentsAction} from "../../../../../core/store/organizations/payment/payments.action";
+import {ServiceLimitsAction} from "../../../../../core/store/organizations/service-limits/service-limits.action";
 
 @Component({
   selector: 'payments-list',
   templateUrl: './payments-list.component.html',
   styleUrls: ['./payments-list.component.scss']
 })
-export class PaymentsListComponent implements OnInit {
+export class PaymentsListComponent implements OnInit, OnDestroy {
 
-  constructor() { }
+  private organization: OrganizationModel;
+  private organizationSubscription: Subscription;
+
+  public payments: PaymentModel[];
+  private paymentsSubscription: Subscription;
+
+  constructor(private i18n: I18n,
+              private router: Router,
+              private store: Store<AppState>) { }
 
   public ngOnInit() {
+    this.subscribeToStore();
+    this.requestData();
+  }
+
+  public subscribeToStore() {
+    this.organizationSubscription = this.store.select(selectOrganizationByWorkspace)
+      .pipe(filter(organization => !isNullOrUndefined(organization)))
+      .subscribe(organization => this.organization = organization);
+
+    this.paymentsSubscription = this.store.select(selectPaymentsByWorkspaceSorted)
+      .pipe(filter(payments => !isNullOrUndefined(payments) && payments.length > 0))
+      .subscribe(payments => this.payments = payments);
   }
 
+  public requestData() {
+    this.store.dispatch(new PaymentsAction.GetPayments({ organizationId: this.organization.id }));
+  }
+
+  public ngOnDestroy(): void {
+    if (this.organizationSubscription) {
+      this.organizationSubscription.unsubscribe();
+    }
+
+    if (this.paymentsSubscription) {
+      this.paymentsSubscription.unsubscribe();
+    }
+  }
+
+  public refreshPayment(paymentId: string) {
+    this.store.dispatch(new PaymentsAction.GetPayment({ organizationId: this.organization.id, paymentId, nextAction: new ServiceLimitsAction.GetServiceLimits({ organizationId: null }) }));
```

Comment:
Why do you pass `organizationId` to `GetPayment` action but do not set it in `GetServiceLimits` action?

Reply:
Because it is later set in the action, I copied that pattern from a different effects (collections). Should I set it in here?

Reply:
It was done this way in other effects because there were some actions that depend on entity ID which is not known at the time when the first action is dispatched in a component. For example when you create a new document with a new link instance, you first need to create a document, obtain its ID, and then use it to create a new link instance.

But in this case it does not make sense to do it in such a complicated way.